### ISSUE TITLE
SimpleTcpServer DataReceived method spawns a new task for each DataReceived event

### ISSUE THREAD

**Issue by TheNybbler** [jchristn/SuperSimpleTcp#160]
The implementation of the `DataReceiver` method in the `SimpleTcpServer` spawns a new Task to call `_events.HandeDataReceived`. this could potentially lead to problems as there is no guarantee that the tasks will run in the correct order (in high data rate scenarios). As each client runs the `DataReceiver` as a separate task, I don't think it is necessary to do this (unless I'm missing something?)

Interestingly, the `SimpleTcpClient` equivalent method does NOT spawn a new Task to call the `_events.HandleDataReceived` method, which is inconsistent.

**Comment by jchristn** [comment]
The client doesn't need to. It's managing exactly one TCP connection. The server manages many which is why a task is spawned for each. I did not want to implement my own scheduler so I left that to be the job of the underlying operating system.

**Comment by TheNybbler** [comment]
But each server client connection already spawns a task to run its own data receiver. I don't see the benefit of then also spawning a task to handle the firing of the data received event.

**Comment by jchristn** [comment]
Those events are fired synchronously. By burying them in a task it allows the server to fire more in parallel.
